client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/CRUDDocumentationIT.java

```diff
@@ -750,6 +761,144 @@ public void onFailure(Exception e) {
         }
     }
 
+    public void testReindex() throws Exception {
+        RestHighLevelClient client = highLevelClient();
+        {
+            String mapping =
+                "\"doc\": {\n" +
+                    "    \"properties\": {\n" +
+                    "      \"user\": {\n" +
+                    "        \"type\": \"text\"\n" +
+                    "      },\n" +
+                    "      \"field1\": {\n" +
+                    "        \"type\": \"integer\"\n" +
+                    "      },\n" +
+                    "      \"field2\": {\n" +
+                    "        \"type\": \"integer\"\n" +
+                    "      }\n" +
+                    "    }\n" +
+                    "  }";
+            createIndex("source1", Settings.EMPTY, mapping);
+            createIndex("source2", Settings.EMPTY, mapping);
+            createPipeline("my_pipeline");
+        }
+        {
+            // tag::reindex-request
+            ReindexRequest request = new ReindexRequest(); // <1>
+            request.setSourceIndices("source1", "source2"); // <2>
+            request.setDestIndex("dest");  // <3>
+            // end::reindex-request
+            // tag::reindex-request-versionType
+            request.setDestVersionType(VersionType.EXTERNAL); // <1>
+            // end::reindex-request-versionType
+            // tag::reindex-request-opType
+            request.setDestOpType("create"); // <1>
+            // end::reindex-request-opType
+            // tag::reindex-request-conflicts
+            request.setConflicts("proceed"); // <1>
+            // end::reindex-request-conflicts
+            // tag::reindex-request-typeOrQuery
+            request.setSourceDocTypes("doc"); // <1>
+            request.setSourceQuery(new TermQueryBuilder("user", "kimchy")); // <2>
+            // end::reindex-request-typeOrQuery
+            // tag::reindex-request-size
+            request.setSize(10); // <1>
+            // end::reindex-request-size
+            // tag::reindex-request-sourceSize
+            request.setSourceBatchSize(100); // <1>
+            // end::reindex-request-sourceSize
+            // tag::reindex-request-pipeline
+            request.setDestPipeline("my_pipeline"); // <1>
+            // end::reindex-request-pipeline
+            // tag::reindex-request-sort
+            request.addSortField("field1", SortOrder.DESC); // <1>
+            request.addSortField("field2", SortOrder.ASC); // <2>
+            // end::reindex-request-sort
+            // tag::reindex-request-script
+            request.setScript(
+                new Script(
+                    ScriptType.INLINE, "painless",
+                    "if (ctx._source.foo == 'bar') {ctx._version++; ctx._source.remove('foo')}",
```

Comment:
I think messing with the `_version` is a fairly rare thing. I think It'd be more normal to, say, add split a field on a regex and stick it into two fields. Or something like that.

Reply:
Ah, I copied this from the other doc. It is to show (I believe) that you can mess with it during `reindex` but not during `update_by_query`. I can change it.

Reply:
I think it is worth changing. Probably in both places to be honest, but just in this place in this PR.